postgres_base_block.py

```diff
@@ -78,26 +77,22 @@ def configure(self, context):
         self.connect()
 
         # create list of column names for insertion validation
-        self._cur.execute("SELECT column_name FROM information_schema.columns \
-                          WHERE table_name = '{}';".format(self.table_name))
+        self._cur.execute("SELECT column_name FROM information_schema.columns "
+                          "WHERE table_name = '{}';".format(self.table_name()))
         self.column_names = [row[0] for row in self._cur]
 
-
     def stop(self):
         self.logger.debug('closing postgres connection...')
         self.disconnect()
         super().stop()
 
     def process_signals(self, signals):
-        self.execute_with_lock(
```

Comment:
Does it make more sense for the `lock` to be outside` of the `retry`? The point of this limit lock is so that if the function if calls takes too long, then we won't continue to queue up input signals.